### ISSUE TITLE
Problems with NIR tutorial

### ISSUE THREAD

**Issue by 0Kevin0** [danforthcenter/plantcv#374]
### Description
Code crashes when I attempt to copy and run the NIR tutorial locally. 

### Details
I am trying to learn how to do things on PlantCV, so I might be ignorant of several details on the project (please be patient). To learn the basics I installed everything and already have been successful on doing the VIS tutorial, first on a local Jupyter notebook with a custom picture and I was also successful on porting it to a script.

When I tried to repeat this with the NIR tutorial, I staarted by downloading the images from your Jupyter notebook (as I lack hardware for now), it didn't work as `bkg_sub_img` and other methods work in "grayscale", I tried to fix this using the `pcv.rgb2gray()` method on the downloaded images and things kind of worked at first. 

The next problem was in the cell 31 method `id_objects,obj_hierarchy = pcv.find_objects(img, edge_masked_img)`, which works fine in your notebook, but in mine shows this error:

> ---------------------------------------------------------------------------
> error                                     Traceback (most recent call last)
> <ipython-input-33-85b96ceb4ef5> in <module>
>       4 #   img - RGB or grayscale image data for plotting
>       5 #   mask - Binary mask used for detecting contours
> ----> 6 id_objects,obj_hierarchy = pcv.find_objects(gray_img1, edge_masked_img)
> 
> ~\Anaconda3\envs\plantcv\lib\site-packages\plantcv\plantcv\find_objects.py in find_objects(img, mask)
>      33     if len(np.shape(ori_img)) == 2:
>      34         ori_img = cv2.cvtColor(ori_img, cv2.COLOR_GRAY2BGR)
> ---> 35     objects, hierarchy = cv2.findContours(mask1, cv2.RETR_TREE, cv2.CHAIN_APPROX_NONE)[-2:]
>      36     for i, cnt in enumerate(objects):
>      37         cv2.drawContours(ori_img, objects, i, (255, 102, 255), -1, lineType=8, hierarchy=hierarchy)
> 
> error: OpenCV(3.4.4) ..\modules\imgproc\src\contours.cpp:195: error: (-210:Unsupported format or combination of formats) [Start]FindContours supports only CV_8UC1 images when mode != CV_RETR_FLOODFILL otherwise supports CV_32SC1 images only in function 'cvStartFindContours_Impl'

I attempted to copy the python script, and I also found that in the script you use the method `kg_sub_thres_img = pcv.threshold.custom_range(bkg_sub_img, [50], [190], channel='gray')
` on a previous segment, and I tried also to use it in an attempt to correct the small errors produced by using the rgb2gray method and then try again the find_objects.

The result was that python returned the message: 

> AttributeError: module 'plantcv.plantcv.threshold' has no attribute 'custom_range'

Which is very strange as the method appears in the Github source, but also crashes in your tutorial Notebook.

### Completion Criteria
a) A small explanation on why find_objects crashed and how to solve this
b) Also an explanation on why custom_range is not usable in the Jupyter Notebook

**Comment by HaleySchuhl** [comment]
Hi @0Kevin0 ,

The tutorial was updated to reflect changes to the master branch but some of these updates (like the addition of the pcv.threshold.custom_range function) were added after the PlantCV v3.2 release, so if you did `pip install PlantCV` or `conda install PlantCV`, this function won't be in your environment. We will have a v3.3 release very soon so this is temporary. This came up in #367 as well, if you replace kg_sub_thres_img = pcv.threshold.custom_range(bkg_sub_img, [50], [190], 'gray')
with bkg_sub_thres_img = cv2.inRange(bkg_sub_img, 50, 190) this ought to get rid of the attribute error. The tutorial notebook is using PyPI so it only has functions in v3.2 since we make new builds with each release. 

I'm unsure what is happening with the line `id_objects,obj_hierarchy = pcv.find_objects(img, edge_masked_img)`. Based on the error message it seems that the image you are using is of an unsupported format. I would suggest to first try updating the way you read your images in. In our example images we already have images in appropriate formats but in the NIR script available on the documentation page we have flags that read in images as grayscale (black and white images are often in RGB formats).  
```
    # Read image (Note: flags=0 indicates to expect a grayscale image)
    img = cv2.imread(args.image, flags=0)

    # Get directory path and image name from command line arguments
    path, img_name = os.path.split(args.image)

    # Read in image which is the pixelwise average of background images
    img_bkgrd = cv2.imread("background_nir_z2500.png", flags=0)
``` 
Another option is trying `img, path, filename = pcv.readimage(args.image, mode="gray")` since the default mode while using `pcv.readimage` is "native". 

Hopefully this helps! Let us know if this works and if you have any other questions.

**Comment by 0Kevin0** [comment]
Thank you Haley, I did in fact installed through conda, so that explains it.

**Comment by nfahlgren** [comment]
Hi @0Kevin0, we merged the new changes yesterday and will work on pushing the new release (3.3.0) to PyPI and Bioconda today. Since releases lag behind the master branch on GitHub we just reconfigured the documentation to default to `stable`, which is the latest release version. That way the default docs should match the latest version on PyPI and Bioconda. If you ever need to, in the bottom right-hand corner of the documentation is a little menu that let's you change the version of the docs you are viewing.
